ct_only.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tessera
```

Comment:
`tessera.Storage` is somewhat independent from CT, but is defined inside ct_only.go. It seems a bit odd. Would it make sense to define NewCertificateTransparencySequencedWriter and convertCTEntry in their own package?

Reply:
That interface is there currently just because there's not a corresponding `tessera.NewSequencedWriter` yet.

Ideally the CT stuff _would_ be in its own package, but to prevent other new use-cases from using these patterns it needs be in here to do "private" stuff internal to `Entry` when setting those marshaling funcs.